server/src/main/java/org/elasticsearch/transport/InboundMessage.java

```diff
@@ -11,13 +11,14 @@
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.common.bytes.ReleasableBytesReference;
 import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.core.AbstractRefCounted;
 import org.elasticsearch.core.IOUtils;
 import org.elasticsearch.core.Releasable;
 
 import java.io.IOException;
 import java.util.Objects;
 
-public class InboundMessage implements Releasable {
+public class InboundMessage extends AbstractRefCounted {
```

Comment:
Can we assert that ref count is greater than 0 when `openOrGetStreamInput` is called?

Reply:
++ done